### ISSUE TITLE
pip is no longer installing

### ISSUE THREAD

**Issue by GarethNelson** [steemit/steem-python#68]
This is due to the weird stuff with setup.py (my own work, intended to allow keeping track of dependencies in only one place).

pip requires an "egginfo" subcommand in setup.py and fails to install without it:
pip.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in /tmp/tmpbpllscr_build/steem/

To reproduce, do "pip install steem" in a clean environment

**Comment by goldibex** [comment]
i actually get a different error:

```
   Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/fh/rzqpp5b1311fksh3hpqmt7v00000gn/T/pip-build-t0ui007v/steem/setup.py", line 8, in <module>
        import package_meta # this is a simple tool for loading metadata from Pipfile and other places
    ModuleNotFoundError: No module named 'package_meta'
```

**Comment by GarethNelson** [comment]
That is a weird one as package_meta is in the same directory as setup.py

**Comment by GarethNelson** [comment]
Some notes:
It seems to work with setuptools version 36.2.7, which is not the default.
I'm going to test upping the version number from python 3.5 to 3.6

**Comment by enki74** [comment]
(steemit_env) telesti@starship:~/python$ pip install -U steem
Collecting steem
  Using cached steem-0.18.103.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-3k9th27l/steem/setup.py", line 10, in <module>
        assert sys.version_info[0] == package_meta.required_python_major and sys.version_info[1] >= package_meta.required_python_minor, "%s requires Python %s or newer" % (package_meta.package_name, package_meta.required_python_ver)
    AssertionError: steem requires Python 3.6 or newer

Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-3k9th27l/steem/

**Comment by GarethNelson** [comment]
"AssertionError: steem requires Python 3.6 or newer"

Are you using python 3.6?

**Comment by enki74** [comment]
I have installed it im not sure if its being used Im a little fuzzy on the environment. Im running ubuntu 16.04
after checking looks like not
(steemit_env) telesti@starship:~/python$ python --version
Python 3.5.2
(steemit_env) telesti@starship:~/python$ python3 --version
Python 3.5.2
how do I change it?

**Comment by enki74** [comment]
(steemit_env) telesti@starship:~/python$ sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.5 1
[sudo] password for telesti: 
update-alternatives: using /usr/bin/python3.5 to provide /usr/bin/python3 (python3) in auto mode
(steemit_env) telesti@starship:~/python$ 
(steemit_env) telesti@starship:~/python$ sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.6 2
update-alternatives: using /usr/bin/python3.6 to provide /usr/bin/python3 (python3) in auto mode
(steemit_env) telesti@starship:~/python$ sudo update-alternatives --config python3
There are 2 choices for the alternative python3 (providing /usr/bin/python3).

  Selection    Path                Priority   Status
------------------------------------------------------------
* 0            /usr/bin/python3.6   2         auto mode
  1            /usr/bin/python3.5   1         manual mode
  2            /usr/bin/python3.6   2         manual mode

Press <enter> to keep the current choice[*], or type selection number: 2
(steemit_env) telesti@starship:~/python$ 
 still showing this 
Press <enter> to keep the current choice[*], or type selection number: 2
(steemit_env) telesti@starship:~/python$ python --version
Python 3.5.2

**Comment by enki74** [comment]
OK tried again now this:

(steemit_env) telesti@starship:~/python$ python3.6 -m pip install -U steem
Collecting steem
  Using cached steem-0.18.103.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-94spzq3n/steem/setup.py", line 8, in <module>
        import package_meta # this is a simple tool for loading metadata from Pipfile and other places
      File "/tmp/pip-build-94spzq3n/steem/package_meta.py", line 10, in <module>
        from setuptools.config import read_configuration
    ModuleNotFoundError: No module named 'setuptools.config'
    Error in sys.excepthook:
    Traceback (most recent call last):
      File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 63, in apport_excepthook
        from apport.fileutils import likely_packaged, get_recent_crashes
      File "/usr/lib/python3/dist-packages/apport/__init__.py", line 5, in <module>
        from apport.report import Report
      File "/usr/lib/python3/dist-packages/apport/report.py", line 30, in <module>
        import apport.fileutils
      File "/usr/lib/python3/dist-packages/apport/fileutils.py", line 23, in <module>
        from apport.packaging_impl import impl as packaging
      File "/usr/lib/python3/dist-packages/apport/packaging_impl.py", line 23, in <module>
        import apt
      File "/usr/lib/python3/dist-packages/apt/__init__.py", line 23, in <module>
        import apt_pkg
    ModuleNotFoundError: No module named 'apt_pkg'
    
    Original exception was:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-94spzq3n/steem/setup.py", line 8, in <module>
        import package_meta # this is a simple tool for loading metadata from Pipfile and other places
      File "/tmp/pip-build-94spzq3n/steem/package_meta.py", line 10, in <module>
        from setuptools.config import read_configuration
    ModuleNotFoundError: No module named 'setuptools.config'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-94spzq3n/steem/

**Comment by E-D-A** [comment]
You are missing setuptools, did you already try installing it?

**Comment by enki74** [comment]
Yes I did but I'll work on it some more! Use pip?

**Comment by enki74** [comment]
tried this now

telesti@starship:~/python$ sudo -H python3.6 -m pip install -U steem
Collecting steem
  Using cached steem-0.18.103.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ynnyueap/steem/setup.py", line 8, in <module>
        import package_meta # this is a simple tool for loading metadata from Pipfile and other places
      File "/tmp/pip-build-ynnyueap/steem/package_meta.py", line 10, in <module>
        from setuptools.config import read_configuration
    ModuleNotFoundError: No module named 'setuptools.config'
    Error in sys.excepthook:
    Traceback (most recent call last):
      File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 63, in apport_excepthook
        from apport.fileutils import likely_packaged, get_recent_crashes
      File "/usr/lib/python3/dist-packages/apport/__init__.py", line 5, in <module>
        from apport.report import Report
      File "/usr/lib/python3/dist-packages/apport/report.py", line 30, in <module>
        import apport.fileutils
      File "/usr/lib/python3/dist-packages/apport/fileutils.py", line 23, in <module>
        from apport.packaging_impl import impl as packaging
      File "/usr/lib/python3/dist-packages/apport/packaging_impl.py", line 23, in <module>
        import apt
      File "/usr/lib/python3/dist-packages/apt/__init__.py", line 23, in <module>
        import apt_pkg
    ModuleNotFoundError: No module named 'apt_pkg'
    
    Original exception was:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ynnyueap/steem/setup.py", line 8, in <module>
        import package_meta # this is a simple tool for loading metadata from Pipfile and other places
      File "/tmp/pip-build-ynnyueap/steem/package_meta.py", line 10, in <module>
        from setuptools.config import read_configuration
    ModuleNotFoundError: No module named 'setuptools.config'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-ynnyueap/steem/

**Comment by E-D-A** [comment]
So if you go into python command line, with "python3.6", and you type "import setuptools.config". I assume you get a similar error? So it means the module is not installed.
If so, please run "pip -V" from linux command line and see what version is returned. Is it pointing to Python3.6? You should also have a pip3 command or even a pip3.6 command, in case the references are not correct.
The issue with the apt_pkg, see this link if it can be of any help:
https://www.mail-archive.com/ubuntu-bugs@lists.ubuntu.com/msg5119661.html

Overall, it might be easier to use Docker to python 3.6 up and running.

**Comment by enki74** [comment]
OK this 
telesti@starship:~$ python3..6 import setuptools.config
python3..6: command not found
telesti@starship:~$ python3.6 import setuptools.config
python3.6: can't open file 'import': [Errno 2] No such file or directory
telesti@starship:~$ pip -V
pip 9.0.1 from /home/telesti/.local/lib/python2.7/site-packages (python 2.7)
telesti@starship:~$ pip3.6 -V
pip3.6: command not found
telesti@starship:~$ pip3 -V
pip 8.1.1 from /usr/lib/python3/dist-packages (python 3.5)
telesti@starship:~$ 


no pip3.6 command 
telesti@starship:~$ pip3.6 install -U steem
pip3.6: command not found

Im confused will installing apt_pkg as described in the link https://www.mail-archive.com/ubuntu-bugs@lists.ubuntu.com/msg5119661.html
fix the setuptools.config module not found?
Not sure what docker to python 3.6 is but looking it up now

**Comment by GarethNelson** [comment]
Please follow these directions:
1 - Download the source tarball from https://www.python.org/downloads/release/python-363/ and use ./configure and "make altinstall"

2 - python3.6 -m pip install -U pip

3 - pip3.6 install -U setuptools

4 - pip3.6 install -U steem

Apologies for all the confusion, there are some features in 3.6 that are now relied upon by the package, the above instructions should work on most mainstream linux distros.

**Comment by GarethNelson** [comment]
With regards to docker, it's a containerisation platform, a full tutorial would be out of scope for this repo but basically you should be able to use one of these images as a base:

https://hub.docker.com/_/python/

Then do "pip3.6 install -U steem" and setup your application code within the Dockerfile.

**Comment by enki74** [comment]
ok got to install ofsteem Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/local/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/local/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/local/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/local/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/local/lib/python3.6/site-packages/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/usr/local/lib/python3.6/site-packages/pip/wheel.py", line 323, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/usr/local/lib/python3.6/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
PermissionError: [Errno 13] Permission denied: '/usr/local/lib/python3.6/site-packages/appdirs.py'

**Comment by GarethNelson** [comment]
You need to do it as root if you're installing globally :)

**Comment by enki74** [comment]
this is whatt i did telesti@starship:~/Downloads/Python-3.6.3$ sudo -H pip3.6 install -U steem

**Comment by GarethNelson** [comment]
If you ran it with sudo i'm at a loss to explain why you're getting permissions errors, but this is not really a steem-python issue as such.

**Comment by enki74** [comment]
hey thanks for your help I seem to have steem installed now running the sample follow.py script getting errors Ill take over to that group!  Thanks a bunch!!! 

elesti@starship:~/Downloads/Python-3.6.3$ sudo -i pip3.6 install -U sqlite3
[sudo] password for telesti: 
Collecting sqlite3
  Downloading sqlite3-99.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-mdtqdvpe/sqlite3/setup.py", line 2, in <module>
        raise RuntimeError("Package 'sqlite3' must not be downloaded from pypi")
    RuntimeError: Package 'sqlite3' must not be downloaded from pypi
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-mdtqdvpe/sqlite3/

**Comment by GarethNelson** [comment]
sqlite3 should be part of the standard library, but if you installed python 3.6 without libsqlite3 installed, it won't compile.

Check your distro's packages for libsqlite3-dev and then reinstall python 3.6

**Comment by warpo** [comment]
I also received some installation issues when installing steem via pip as a dependency for myownproject. Note I'm working in Ubuntu 16.04 LTS. My OS does not package Python 3.6 as default so I had to install manually as this is a requirement for steem. Here are steps I followed.
`sudo apt install python3-setuptools libsqlite3-dev`
`cd ~/Downloads`
`wget https://www.python.org/ftp/python/3.6.3/Python-3.6.3.tar.xz`
`tar xvf Python-3.6.3.tar.xz`
`cd Python-3.6.3`
`./configure --enable-loadable-sqlite-extensions`
`make`
`sudo make install`
This will fetch and build Python3.6.3 on your local system. You can check `python --version` to ensure you have the right Python. Then in the directory for myownproject I create a virtual environment.
`python3 -m venv .` Note the dot (.) represents my location as within the myownproject dir.
Then I activate the virtualenv `source bin/activate` and update some packages
`pip install -U pip setuptools`
Finally I install steem
`pip install steem`
However, at this point if you run `steempy -h` you will get an error complaining about the toml package being required to be 
> Requirement.parse('toml==0.9.3.1')

but the latest in pip is only 0.9.3. Therefore, I believe the fix is updating the requirements files to correct this. I'll put in a pull request for this.

**Comment by bennierex** [comment]
The toml issue is still existent in the latest steem-python version. Suggesting updating the requirements to toml==0.9.4 (assuming it has no backwards-incompatible changes).

I also encountered the `ModuleNotFoundError: No module named 'setuptools.config'` error, wich I resolved by running `pip install --upgrade setuptools` before installing steem-python.

**Comment by kirkins** [comment]
I'm facing the same error installing with pip

**Comment by kpekarov** [comment]
Same here. 
I can't install steem on server (ubuntu 16.04) and on my dev PC (also 16.04). 

Upgrading all just because of steem lib is not an option. Using docker also make development much more complex and slow.

**Comment by cyon1c** [comment]
This issue is fixed at commit [1d7e7f3](https://github.com/steemit/steem-python/commit/1d7e7f3a62ec180a986f0d4504939e822bc194be). A new version will be uploaded to PyPi in short order.
